### PR TITLE
Expose `max_tx_size` API

### PR DESCRIPTION
**Pull request by unseddd**
Related to grin-wallet [#495](https://github.com/mimblewimble/grin-wallet/pull/495)

### CODE REVIEW

**Comment by antiochp** [comment]
Agree that we should have a limit on slate sizes. 
But I'm not convinced `max_block_size` is the correct thing to base this off.

**Comment by unseddd** [comment]
> But I'm not convinced max_block_size is the correct thing to base this off.

Understand, response is in [#495](https://github.com/mimblewimble/grin-wallet/pull/495).

**Comment by antiochp** [comment]
Maybe one option here would be to expose a new `max_tx_size()` (this can just be based off max_block_size internally).
This way the wallet would be using an api less likely to change unexpectedly.

May need to do additional calcs on the wallet side but this would be fine here.

**Comment by unseddd** [comment]
> Even better if we could expose this as `global::max_tx_size()` and avoid needing to `require p2p` from wallet.

Exposing as `global::max_tx_size()` creates a cyclic dependency, so instead chose to expose as `api::max_tx_size()`.

Seems a logical place for the function, `api` already requires the needed sub-crates, and `api` is already required by `grin_wallet::util`. WDYT?

File: api/src/lib.rs [hunk 1]
```diff
@@ -56,3 +56,9 @@ pub use crate::rest::*;
 pub use crate::router::*;
 pub use crate::types::*;
 pub use crate::web::*;
+
+/// Get max transaction size
+pub fn max_tx_size() -> u64 {
+	p2p::msg::max_block_size()
+		- std::mem::size_of_val(&core::core::block::BlockHeader::default()) as u64
```
Comment (antiochp) [inline api/src/lib.rs:63]: This does not necessarily correspond to a serialized size in bytes. Its simply the internal representation, which will likely be different.

**Pull request merged**